### ISSUE TITLE
Avoid problems when file_get_contents() is not available

### ISSUE THREAD

**Issue by exside** [modxcms/imageSlim#1]
At the moment I don't really use it =), but I just know of some servers disabling allow_url_fopen (had problems with my own code), so here's what I usually do:
1. check for cURL and use $modx functionality to fetch what i need
2. check for file_get_contents() working and do it that way...

you probably need to adapt the code a bit, but it could get you rid of that requirement =D

``` php
if (function_exists('curl_init')) {
    $modx->getService('getfile','rest.modRestCurlClient');
    $response = $modx->getfile->request($url);
} else if ( function_exists('file_get_contents') && ini_get('allow_url_fopen') ) {
    $response = @file_get_contents($url);
} else {
    $modx->log(modX::LOG_LEVEL_ERROR, 'Neither cURL nor file_get_contents() is available to fetch remote data...talk to your hoster or get a better one');
}
```

**Comment by oo12** [comment]
Thanks, this is a good idea.  I'll incorporate it into the pl release.

**Comment by oo12** [comment]
Fixed in v1.0.0-pl.  cURL is now used for remote images, since it's more likely to be available.  I also added local caching and a maximum download time limit for remote images, which should eliminate the major performance issues with remote images.  Now they only have to be downloaded once, even if their size gets changed later.
